Xamarin.Forms.Core/MediaElement.cs

```diff
@@ -0,0 +1,267 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Xamarin.Forms.Platform;
+
+namespace Xamarin.Forms
+{
+	[RenderWith(typeof(_MediaElementRenderer))]
+	public sealed class MediaElement : View
+	{
+		public static readonly BindableProperty AreTransportControlsEnabledProperty =
+		  BindableProperty.Create(nameof(AreTransportControlsEnabled), typeof(bool), typeof(MediaElement), false);
+
+		public static readonly BindableProperty AspectProperty =
+		  BindableProperty.Create(nameof(Aspect), typeof(Aspect), typeof(MediaElement), Aspect.AspectFit);
+
+		public static readonly BindableProperty AutoPlayProperty =
+		  BindableProperty.Create(nameof(AutoPlay), typeof(bool), typeof(MediaElement), true);
+
+		public static readonly BindableProperty BufferingProgressProperty =
+		  BindableProperty.Create(nameof(BufferingProgress), typeof(double), typeof(MediaElement), 0.0);
+
+		public static readonly BindableProperty IsLoopingProperty =
+		  BindableProperty.Create(nameof(IsLooping), typeof(bool), typeof(MediaElement), false);
+
+		public static readonly BindableProperty KeepScreenOnProperty =
+		  BindableProperty.Create(nameof(KeepScreenOn), typeof(bool), typeof(MediaElement), false);
+
+		public static readonly BindableProperty SourceProperty =
+		  BindableProperty.Create(nameof(Source), typeof(Uri), typeof(MediaElement));
```

Comment:
This will probably require a MediaSource class similar to the ImageSource class, with the appropriate subclasses (FileMediaSource, StreamMediaSource, UriMediaSource and their respective type converters).